### ISSUE TITLE
highlightFirstSuggestion bugs

### ISSUE THREAD

**Issue by fanfoorik** [moroshko/react-autosuggest#489]
[Bugs example(codepen)](https://codepen.io/anon/pen/YYPmNq)

I have "highlighted:" label and simple input element on a page.

Set highlightFirstSuggestion to true.
Provide onSuggestionHighlighted method that returns suggested object and sets highlighted label value to suggested.name property.

* First bug:
  1. Focus on the input field
  2. Type "c". First highlighted item is "C", - "highlighted: C". Its ok!
  3. Continue typing "+" to be "c+". First highlighted item is "C++", but highlighted label still "C" - "highlighted: C". not ok!

 **Observed behaviour:** "highlighted: C"
**Expected behaviour:** "highlighted: C++"

* Second bug:
  1. Focus on the input field
  2.  Type "c". First highlighted item is "C", - "highlighted: C". Its ok!
  3. Move your mouse on other suggested items. onSuggestionHighlighted doesnt work properly and highlighted label is still "C" - "highlighted: C". not ok!

 **Observed behaviour:** "highlighted: C".
**Expected behaviour:** "highlighted: C++" or "highlighted: C#" or "highlighted: Clojure"... up to what element you hover on".

**Comment by mdelagrammatikas** [comment]
I'm seeing the behavior described in the first bug as well. The `onSuggestionHighlighted` method is not being triggered on subsequent key presses even though the list is filtered and the first suggestion is now something else. Seems like it's only triggered when value goes from nil -> something and vice versa.

I've fixed it by setting the value to '' and then back to the previous value, which is a little weird but works for the time being.

```
onSuggestionsFetchRequested = async ({ value }: { value: string }) => {
  await this.setState({
    value: '',
    suggestions: await this.props.fetchSuggestions(value)
  })

  this.setState({value: value})
}
```

**Comment by moroshko** [comment]
@fanfoorik Thanks for reporting!

Both bugs should be fixed in [v9.3.4](https://github.com/moroshko/react-autosuggest/releases/tag/v9.3.4).
